pkg/sharing/secret_reconciler_test.go

```diff
@@ -0,0 +1,321 @@
+				Name:      "test-secret",
+				Namespace: "test-source",
+			},
+			Spec: sg2v1alpha1.SecretExportSpec{
+				ToNamespaces: []string{"*"},
+			},
+		}
+		return
+	}
+
+	reconcilersUnderTest := func(objects ...runtime.Object) (secretExportReconciler *sharing.SecretExportReconciler, secretReconciler *sharing.SecretReconciler, k8sClient client.Client) {
+		secretExports := sharing.NewSecretExportsWarmedUp(sharing.NewSecretExports(testLogr))
+		k8sClient = fakeClient.NewFakeClient(objects...)
+		secretExportReconciler = sharing.NewSecretExportReconciler(k8sClient, secretExports, testLogr)
+		secretReconciler = sharing.NewSecretReconciler(k8sClient, secretExports, testLogr)
+		secretExports.WarmUpFunc = secretExportReconciler.WarmUp
+
+		return
+	}
+	t.Run("star export goes to all namespaces", func(t *testing.T) {
+		sourceSecret, placeholderSecret1, placeholderSecret2, secretExport := resourcesUnderTest()
+		secretExportReconciler, secretReconciler, k8sClient := reconcilersUnderTest(&sourceSecret, &placeholderSecret1, &placeholderSecret2, &secretExport)
+
+		reconcileObject(t, secretExportReconciler, &secretExport)
+		reconcileObject(t, secretReconciler, &placeholderSecret1)
+		reconcileObject(t, secretReconciler, &placeholderSecret2)
+
+		reload(t, &placeholderSecret1, k8sClient)
+		reload(t, &placeholderSecret2, k8sClient)
+
+		assert.Equal(t, sourceSecret.Data[".dockerconfigjson"], placeholderSecret1.Data[".dockerconfigjson"])
+		assert.Equal(t, sourceSecret.Data[".dockerconfigjson"], placeholderSecret2.Data[".dockerconfigjson"])
+	})
+
+	t.Run("specific export goes only to specific namespace", func(t *testing.T) {
+		sourceSecret, placeholderSecret1, placeholderSecret2, secretExport := resourcesUnderTest()
+		secretExport.Spec = sg2v1alpha1.SecretExportSpec{ToNamespaces: []string{placeholderSecret1.Namespace}}
+		secretExportReconciler, secretReconciler, k8sClient := reconcilersUnderTest(&sourceSecret, &placeholderSecret1, &placeholderSecret2, &secretExport)
+		reconcileObject(t, secretExportReconciler, &secretExport)
+		reconcileObject(t, secretReconciler, &placeholderSecret1)
+		reconcileObject(t, secretReconciler, &placeholderSecret2)
+
+		// placeholder secret2 should have its original contents for auths and a helpful status message
+		originalPlaceholder2Data := make([]byte, len(placeholderSecret2.Data[".dockerconfigjson"]))
+		copy(originalPlaceholder2Data, placeholderSecret2.Data[".dockerconfigjson"])
+
+		reload(t, &placeholderSecret1, k8sClient)
+		reload(t, &placeholderSecret2, k8sClient)
+
+		assert.Equal(t, sourceSecret.Data[".dockerconfigjson"], placeholderSecret1.Data[".dockerconfigjson"])
+		assert.Equal(t, originalPlaceholder2Data, placeholderSecret2.Data[".dockerconfigjson"])
+		assert.NotEqual(t, placeholderSecret1.Data[".dockerconfigjson"], placeholderSecret2.Data[".dockerconfigjson"])
+	})
+
+}
+
+func Test_SecretReconciler_updatesStatus(t *testing.T) {
+	sg2v1alpha1.AddToScheme(scheme.Scheme)
```

Comment:
i think add to schema should happen in an init() for the test package. otherwise each test keeps on adding same thing.

Reply:
leaving open as it may make sense to centralize this along with instantiation of test fixture objects.